path_2/path_1.py

```diff
@@ -1,13 +1,26 @@
+from functools import wraps, lru_cache
+
+#Python3.2+
+# @lru_cache(maxsize=32)
+# def get_logger():
+#     return [1, 2, 3] * 2
+
+
+#Python2
 
 def once(f):
-    def main(*args, **kwargs):
-        result = f(*args, **kwargs)
 
-        def wrapper():
+    memo = {}
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        try:
+            return memo[args]
+        except KeyError:
+            result = f(*args, **kwargs)
```

Comment:
Generally speaking, it is not recommended to use exception for control flow: https://wiki.c2.com/?DontUseExceptionsForFlowControl

You can just check key existence in dictionary.

Reply:
Thanks! Fixed.